### PR TITLE
Prune QR daily salts on a timer, not only when someone scans

### PR DESCRIPTION
**Pull request by nyagrodha**
Pruning ran only inside `recordScan`, so it stopped whenever scanning stopped. That failed in production within two days.

On 2026-08-13 the **2026-08-11 salt was still present**, because nothing had visited `/WillyStCo-op` since Aug 11 — leaving those two visitors re-linkable past the 48h window the entire unlinkability argument rests on.

The mistake was treating an idle URL as the edge case. For a QR on a wall it is the *normal* case, so the condition that disabled pruning is the condition the feature usually operates under. CodeRabbit flagged this on #94 and I documented it rather than fixing it; production then demonstrated it.

## What changed

- **`scripts/prune-qr-salts.ts`** — scheduled entry point, run from a systemd timer. This is what now guarantees the window. Exits non-zero on failure so the timer surfaces it; for this job, silent failure looks identical to success.
- **`pruneSalts()`** and one shared `SALT_PRUNE_SQL`, so the request path and the timer cannot drift apart.
- **`saltCutoffDay()`** computes the boundary in TypeScript from UTC rather than in SQL from `CURRENT_DATE`, which follows the database session's timezone and would shift the window by a day — the same trap the report query hit.

## The request-path prune is kept, deliberately

Not an oversight. The two mechanisms fail independently: a dead timer is covered by traffic, an idle URL is covered by the timer. The cost is one indexed `DELETE` against a table holding at most two rows. For a property the design rests on, that is worth paying twice.

## Verification

34 tests pass, including the boundary cases that matter: the cutoff is yesterday so today and yesterday survive; a two-day-old salt falls before it; UTC late-evening does not roll it early; month and year boundaries.

Spec updated — the section that documented this as an accepted limitation now records that it failed and how.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved QR scan security-data cleanup by consistently removing expired salts based on UTC time.
  * Ensured cleanup continues during periods with no QR scan activity.
  * Improved handling across month and year boundaries to prevent premature or delayed removal.

### CODE REVIEW

File: docs/superpowers/specs/2026-08-11-coop-qr-scan-counting-design.md [hunk 1]
```diff
@@ -125,21 +125,34 @@ this machinery.
 On the first visit of a UTC day, generate 32 random bytes and
 `INSERT ... ON CONFLICT (day) DO NOTHING`, then `SELECT` the winner — the
 conflict path is what makes two simultaneous first-visits agree on one salt
-instead of racing. The same code path prunes:
-`DELETE FROM fresh_qr_salts WHERE day < CURRENT_DATE - 1`.
+instead of racing. That path also prunes opportunistically, using the same
+`SALT_PRUNE_SQL` (`DELETE FROM fresh_qr_salts WHERE day < $1`) the timer runs,
+parameterised with `saltCutoffDay()` — see "Pruning runs on a timer" below.
```
[the remark this answers is not in the document]
Comment (nyagrodha) [reply docs/superpowers/specs/2026-08-11-coop-qr-scan-counting-design.md:130]: fix this @copilot
